library/core/src/num/nonzero.rs

```diff
@@ -895,3 +895,39 @@ macro_rules! nonzero_unsigned_is_power_of_two {
 }
 
 nonzero_unsigned_is_power_of_two! { NonZeroU8 NonZeroU16 NonZeroU32 NonZeroU64 NonZeroU128 NonZeroUsize }
+
+macro_rules! nonzero_min_max {
+    ( $( $MinVal:expr , $Ty: ident($Int: ty); )+ ) => {
+        $(
+            impl $Ty {
+                #[unstable(feature = "nonzero_max", issue = "89065")]
+                #[doc = concat!("The maximum value for a`", stringify!($Ty), "` is the same as `", stringify!($Int), "`")]
+                #[doc = concat!("assert_eq!(", stringify!($Ty), "::MAX, ", stringify!($Int), "::MAX);")]
+                // SAFETY: Since the MAX value, for any supported integer type, is greater than 0, the MAX will always be non-zero.
+                pub const MAX : $Ty = unsafe { $Ty::new_unchecked(<$Int>::MAX) };
```

Comment:
```suggestion
                pub const MAX : $Ty = $Ty::new(<$Int>::MAX).unwrap();
```
Since we're in a const context this is guaranteed to get evaluated (and potentially fail if you pass in a zero after all) at compile time. No need for unsafe.

Same for the min case below. Unfortunately `unwrap_or` isn't const yet or we could get rid of the `$MinVal` input entirely.

Reply:
I agree! I really wanted to get rid of that $MinVal, but couldn't find a clean way. Fixed the unsafe.
591b89dbd58d890f2e13fadbcfc78f590ec5da4b